.github/workflows/functional-baremetal.yaml

```diff
@@ -0,0 +1,96 @@
+        include:
+          - name: "xena"
+            openstack_version: "stable/xena"
+            ubuntu_version: "20.04"
+          - name: "wallaby"
+            openstack_version: "stable/wallaby"
+            ubuntu_version: "20.04"
+          - name: "victoria"
+            openstack_version: "stable/victoria"
+            ubuntu_version: "20.04"
+          - name: "ussuri"
+            openstack_version: "stable/ussuri"
+            ubuntu_version: "18.04"
+          - name: "train"
+            openstack_version: "stable/train"
+            ubuntu_version: "18.04"
+    runs-on: ubuntu-${{ matrix.ubuntu_version }}
+    name: Deploy OpenStack ${{ matrix.name }} with Ironic and run baremetal acceptance tests
+    steps:
+      - name: Checkout Gophercloud
+        uses: actions/checkout@v2
+      - name: Deploy devstack
+        uses: EmilienM/devstack-action@v0.6
+        with:
+          branch: ${{ matrix.openstack_version }}
+          conf_overrides: |
+            enable_plugin ironic https://opendev.org/openstack/ironic ${{ matrix.openstack_version }}
+            disable_service cinder c-sch c-api c-vol c-bak
+            disable_service ovn ovn-controller ovn-northd q-ovn-metadata-agent
+            enable_service q-svc q-agt q-dhcp q-l3 q-meta
+            LIBS_FROM_GIT=pyghmi,virtualbmc
+            FORCE_CONFIG_DRIVE=True
+            Q_AGENT=openvswitch
+            Q_ML2_TENANT_NETWORK_TYPE=vxlan
+            Q_ML2_PLUGIN_MECHANISM_DRIVERS=openvswitch
+            DEFAULT_INSTANCE_TYPE=baremetal
+            OVERRIDE_PUBLIC_BRIDGE_MTU=1400
+            VIRT_DRIVER=ironic
+            BUILD_TIMEOUT=1800
+            SERVICE_TIMEOUT=90
+            GLANCE_LIMIT_IMAGE_SIZE_TOTAL=5000
+            Q_USE_SECGROUP=False
+            API_WORKERS=1
+            IRONIC_BAREMETAL_BASIC_OPS=True
+            IRONIC_BUILD_DEPLOY_RAMDISK=False
+            IRONIC_AUTOMATED_CLEAN_ENABLED=False
+            IRONIC_CALLBACK_TIMEOUT=600
+            IRONIC_DEPLOY_DRIVER=ipmi
+            IRONIC_INSPECTOR_BUILD_RAMDISK=False
+            IRONIC_RAMDISK_TYPE=tinyipa
+            IRONIC_TEMPEST_BUILD_TIMEOUT=720
+            IRONIC_TEMPEST_WHOLE_DISK_IMAGE=False
+            IRONIC_VM_COUNT=1
+            IRONIC_VM_EPHEMERAL_DISK=1
+            IRONIC_VM_LOG_DIR=/opt/stack/new/ironic-bm-logs
+            IRONIC_VM_SPECS_RAM=1024
+            IRONIC_DEFAULT_DEPLOY_INTERFACE=direct
+            IRONIC_ENABLED_DEPLOY_INTERFACES=direct
+            SWIFT_ENABLE_TEMPURLS=True
+            SWIFT_TEMPURL_KEY=secretkey
+          enabled_services: 'ir-api,ir-cond,s-account,s-container,s-object,s-proxy'
+      - name: Checkout go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15'
+      - name: Run Gophercloud acceptance tests
+        run: ./script/acceptancetest
+        env:
+          DEVSTACK_PATH: ${{ github.workspace }}/devstack
+          ACCEPTANCE_TESTS_FILTER: '^.*baremetal/v1.*$'
```

Comment:
There are [other acceptance tests for BM](https://github.com/gophercloud/gophercloud/tree/master/acceptance/openstack/baremetal). Are you intentionally skipping them?

Note that the `noauth` tests probably won't work against the deployed devstack, you may want to update the regex to skip them, see https://github.com/gophercloud/gophercloud/pull/2268/commits/6bb11252d7cb5f6b979fa90643eb3f283244d1bc

Reply:
I'll update it.

Reply:
@mandre FYI this didn't change anything:

```
=== RUN   TestPortsUpdate
    conditions.go:81: this test requires Ironic using http_basic, set OS_USERNAME, OS_PASSWORD and IRONIC_ENDPOINT
--- SKIP: TestPortsUpdate (0.00s)
PASS
ok  	github.com/gophercloud/gophercloud/acceptance/openstack/baremetal/httpbasic	0.004s
```

Reply:
I suggest we move on from now and we'll revisit later to enable these tests by setting the env vars.

Reply:
Note you then need to use grep with the perl regex for the negative lookahead to work, see https://github.com/gophercloud/gophercloud/commit/6bb11252d7cb5f6b979fa90643eb3f283244d1bc#diff-bcc2d3c788591acc07e26f74aa38c998fb12c07f2ad1bdd9ecaeea46a8705f9dR24